anomalib/data/utils/augmenter.py

```diff
@@ -38,7 +38,15 @@ class Augmenter:
         noise. If not specified, random noise will be used instead.
     """
 
-    def __init__(self, anomaly_source_path: Optional[str] = None):
+    def __init__(
+        self,
+        anomaly_source_path: Optional[str] = None,
+        p_anomalous: float = 0.5,
+        beta: Union[float, Tuple[float, float]] = (0.2, 1.0),
+    ):
```

Comment:
since you're refactoring, could we enjoy the opportunity to add a seed parameter here?
```suggestion
    def __init__(
        self,
        anomaly_source_path: Optional[str] = None,
        p_anomalous: float = 0.5,
        beta: Union[float, Tuple[float, float]] = (0.2, 1.0),
        seed: Optional[int] = None,
    ):
        self.seed = seed or random.randrange(sys.maxsize)
        self.random_generator = random.Random(self.seed)
```

`sys` was not imported!

i found this `random.randrange(sys.maxsize)` [here](https://stackoverflow.com/a/5012617/9582881)

if accepted then all the `random.func_name()` must be replaced by `self.random_generator.func_name()`

---

There is a call to `np.random.choice`, which i think could be replaced by `Random.sample`, giving 

```
# Random.sample() is *without* replacement
aug_ind = np.array(self.random_generator.sample(range(len(self.augmenters)), k=3))
# instead of
# aug_ind = np.random.choice(np.arange(len(self.augmenters)), 3, replace=False)

```